content/client-lib-development-guide/features.textile

```diff
@@ -1697,13 +1734,42 @@ class RealtimeChannel:
   unsubscribe(String, (Message) ->) // RTL8a
   setOptions(options: ChannelOptions) => io // RTL16
 
+class BatchOperations:
+  publish([String], [Message]) => BatchResult<BatchPublishResponse> // BO2a
+  publish(String, [Message]) => BatchResult<BatchPublishResponse> // BO2a
+  publish([String], Message) => BatchResult<BatchPublishResponse> // BO2a
+  publish([BatchSpec]) => BatchResult<BatchPublishResponse> // BO2a
+  getPresence([String]) => BatchResult<BatchPresenceResponse> // BO2b
+
+class BatchResult<T>:
```

Comment:
This looks like a generic where batch result is an array of type T. 
I'm not sure how a programming language which doesn't have generics (like php) would implement this.

Generics are still very, very new in Go and the release notes advising using caution when deploying code containing generics to production (https://go.dev/doc/go1.18) 

Will languages like PHP (and in the immediate short term Go) be able to make their own implementation choices here?

Reply:
Yes, generics are used elsewhere in the IDL and I believe the correct implementation in a language without generics here would be to make `BatchPublishResult` and `BatchPresenceResult`, but that would be up to the implementer.